pkg/operator/ceph/object/objectstore.go

```diff
@@ -782,22 +782,25 @@ func createSimilarPools(ctx *Context, pools []string, clusterSpec *cephv1.Cluste
 	return nil
 }
 
-func createRGWPool(ctx *Context, clusterSpec *cephv1.ClusterSpec, poolSpec cephv1.PoolSpec, pgCount, ecProfileName, pool string) error {
+func createRGWPool(ctx *Context, clusterSpec *cephv1.ClusterSpec, poolSpec cephv1.PoolSpec, pgCount, ecProfileName, requestedName string) error {
 	// create the pool if it doesn't exist yet
-	name := poolName(ctx.Name, pool)
-	if poolDetails, err := cephclient.GetPoolDetails(ctx.Context, ctx.clusterInfo, name); err != nil {
+	pool := cephv1.NamedPoolSpec{
+		Name:     poolName(ctx.Name, requestedName),
+		PoolSpec: poolSpec,
+	}
```

Comment:
Is it necessary to set the `Name` here for RGW pools?

Reply:
Rook's internal pool creation API takes a `NamedPoolSpec` type instead of having a separate name from the pool spec. From rgw's perspective there is no change in behavior, just consolidation of parameters to create the pool.